src/style/style.js

```diff
@@ -1554,9 +1554,13 @@ class Style extends Evented {
             // Updating fog
             const fog = this.fog;
             const currSpec = fog.get();
+
+            // empty object should pass through to set default values
+            if (Object.keys(fogOptions).length === 0) fog.set(fogOptions);
```

Comment:
Could we have a render test for this code path? Based on our conversation, it'd be something like:
```
map.setFog({
    'color': '#0F2127',
    'high-color': '#000',
    'horizon-blend': 0.5,
    'space-color': '#000'
})
map.setFog({})
```

Reply:
Ok sounds good, Yes quickly confirmed that currently in this use-case, the fog isn't changed on the map.setFog({}) call